spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.maxAttempts")
+          .description("Max attempts of retries on unhandled controller errors.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
+          .build();
+
+  public static final ConfigOption<Long> MAX_RETRY_ATTEMPT_ON_KUBE_SERVER_FAILURE =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.maxRetryAttemptOnKubeServerFailure")
+          .description(
+              "Maximal number of retry attempts of requests to k8s server upon "
+                  + "response 429 and 5xx. This would be performed on top of k8s client "
+                  + "spark.kubernetes.operator.retry.maxAttempts. ")
+          .defaultValue(3L)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Long> RETRY_ATTEMPT_AFTER_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.attemptAfterSeconds")
+          .description(
+              "Default time (in seconds) to wait till next request. This would be used if "
+                  + "server does not set Retry-After in response.")
+          .defaultValue(1L)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Long> MAX_RETRY_ATTEMPT_AFTER_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.maxAttemptAfterSeconds")
+          .description("Maximal time (in seconds) to wait till next request.")
+          .defaultValue(15L)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Long> STATUS_PATCH_MAX_RETRY =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.maxStatusPatchAttempts")
+          .description(
+              "Maximal number of retry attempts of requests to k8s server for resource "
+                  + "status update. This would be performed on top of k8s client "
+                  + "spark.kubernetes.operator.retry.maxAttempts to overcome potential "
+                  + "conflicting update on the same SparkApplication. ")
+          .defaultValue(3L)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Long> SECONDARY_RESOURCE_CREATE_MAX_ATTEMPTS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.secondaryResourceCreateMaxAttempts")
+          .description(
+              "Maximal number of retry attempts of requesting secondary resource for Spark "
+                  + "application. This would be performed on top of k8s client "
+                  + "spark.kubernetes.operator.retry.maxAttempts to overcome potential "
+                  + "conflicting reconcile on the same SparkApplication. ")
+          .defaultValue(3L)
+          .typeParameterClass(Long.class)
```

Comment:
BTW, please reformat all `ConfigOption` builder pattern in this file consistently.

For example, in a standard form, `typeParameterClass` should be defined before `defaultValue`.

Reply:
Thanks for the advice!